src/Symfony/Component/PropertyInfo/WriteMutator.php

```diff
@@ -0,0 +1,80 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\PropertyInfo;
+
+/**
+ * Writes mutator tell how to write to a property.
+ *
+ * @author Joel Wurtz <jwurtz@jolicode.com>
+ */
+final class WriteMutator
+{
+    public const TYPE_METHOD = 1;
+    public const TYPE_PROPERTY = 2;
+    public const TYPE_ARRAY_DIMENSION = 3;
+    public const TYPE_ADDER_REMOVER_METHOD = 4;
+    public const TYPE_CONSTRUCTOR = 5;
+
+    private $type;
+    private $name;
+    private $adderName;
+    private $removerName;
+    private $private;
+    private $constructorParameter;
+    private $static;
+
+    public function __construct(int $type, ?string $name, ?string $adderName, ?string $removerName, bool $private = false, bool $static = false, \ReflectionParameter $constructorParameter = null)
+    {
+        $this->type = $type;
+        $this->name = $name;
+        $this->adderName = $adderName;
+        $this->removerName = $removerName;
+        $this->private = $private;
+        $this->constructorParameter = $constructorParameter;
+        $this->static = $static;
+    }
+
+    public function getType(): int
+    {
+        return $this->type;
+    }
+
+    public function getName(): ?string
+    {
+        return $this->name;
+    }
+
+    public function getAdderName(): ?string
+    {
+        return $this->adderName;
+    }
+
+    public function getRemoverName(): ?string
+    {
+        return $this->removerName;
+    }
+
+    public function isPrivate(): bool
+    {
+        return $this->private;
+    }
+
+    public function isStatic(): bool
+    {
+        return $this->static;
```

Comment:
is this applicable for methods as well, or only for the property? i think we should throw an exception when we are a type that does not have it.

Reply:
applicable for boths, so no need for exception

Reply:
it makes no sense for the constructor type.

i'd love to make these methods strict to find programming errors early when using this.